File: stack/java/BefittingBrackets.java
import java.util.ArrayDeque;
import java.util.Deque;

public final class BefittingBrackets {
  private BefittingBrackets() {
  }

  public static boolean check(String s) {
    Deque<Character> stack = new ArrayDeque<>();
    for (char ch : s.toCharArray()) {
      if (ch == '(' || ch == '[' || ch == '{') {
        stack.push(ch);
      } else if (ch == ')' || ch == ']' || ch == '}') {
        if (stack.isEmpty() || stack.pop() != matching(ch)) {
          return false;
        }
      }
    }
    return stack.isEmpty();
  }

  private static char matching(char closing) {
    return switch (closing) {
      case ')' -> '(';
      case ']' -> '[';
      case '}' -> '{';
      default -> '?';
    };
  }
}

File: stack/js/befittingBrackets.js
const befittingBrackets = (str) => {
    const stack = []

    const brackets = {
        '(': ')',
        '[': ']',
        '{': '}'
    }

    for(let char of str){
        if(char in brackets){
            stack.push(char)
        }else{
            if(stack.length > 0 && stack[stack.length -1] !== char){
               stack.pop()
            }else{
                return false
            }
        }
    }
    return stack.length === 0
}